frontend/src/routes/[[lang=locale]]/candidate/(protected)/questions/+page.svelte

```diff
@@ -5,103 +5,88 @@
-
-  let questions = get(questionsStore) ?? [];
-
-  let opinionQuestionsLeft: number | undefined;
-  nofUnansweredOpinionQuestionsStore?.subscribe((value) => {
-    opinionQuestionsLeft = value;
-  });
-
-  let opinionQuestionsFilled: boolean | undefined;
-  opinionQuestionsFilledStore?.subscribe((value) => {
-    opinionQuestionsFilled = value;
-  });
-
-  let basicInfoFilled: boolean | undefined;
-  basicInfoFilledStore?.subscribe((value) => {
-    basicInfoFilled = value;
-  });
-
-  $: answers = $answersStore;
-
-  let questionsByCategory: Record<string, Array<Question>>;
+  let questionsByCategory: Record<string, Array<QuestionProps>> | undefined;
 
   let loading = true;
   let unansweredCategories: Array<string> | undefined;
 
-  $: questionsByCategory = Object.values(questions).reduce(
+  $: questionsByCategory = $opinionQuestions?.reduce(
     (acc, question) => {
       if (!question.category) {
         return acc;
       }
       const category = question.category.name;
-      if (!acc[category]) {
+      if (acc && !acc[category]) {
         acc[category] = [];
       }
-      acc[category].push(question);
+      if (acc) {
+        acc[category].push(question);
+      }
       return acc;
     },
     {} as typeof questionsByCategory
   );
 
   $: {
-    if (questions) {
+    if ($opinionQuestions) {
       loading = true;
 
-      if (answers) {
+      if ($opinionAnswers) {
         loading = false;
-        unansweredCategories = Object.keys(questionsByCategory).filter(
-          (category) => !questionsByCategory[category].every((question) => answers?.[question.id])
-        );
+        if (questionsByCategory) {
+          unansweredCategories = Object.keys(questionsByCategory).filter(
+            (category) =>
+              !questionsByCategory[category].every((question) => $opinionAnswers?.[question.id])
+          );
+        }
       }
     }
   }
 
-  $: nextUnansweredQuestion = Object.values(questions).find((question) => !answers?.[question.id]);
+  $: nextUnansweredQuestion = ($opinionQuestions ?? []).find(
+    (question) => !$opinionAnswers?.[question.id]
+  );
 </script>
 
-{#if answers && !questionsLocked && Object.entries(answers).length === 0}
+{#if $opinionAnswers && !$questionsLocked && Object.entries($opinionAnswers).length === 0}
   <QuestionsStartPage />
 {:else}
   <BasicPage
     title={$t('candidateApp.questions.title')}
-    progress={$progressStore?.progress}
-    progressMax={$progressStore?.max}>
-    <Warning display={!!questionsLocked} slot="note">
+    progress={$progress?.progress}
+    progressMax={$progress?.max}>
+    <Warning display={!!$questionsLocked} slot="note">
       <p>{$t('candidateApp.questions.editingAllowedNote')}</p>
-      {#if !opinionQuestionsFilled || !basicInfoFilled}
+      {#if ($unansweredOpinionQuestions && $unansweredOpinionQuestions.length !== 0) || ($unansweredRequiredInfoQuestions && $unansweredRequiredInfoQuestions.length !== 0)}
```

Comment:
I think `#if ($unansweredOpinionQuestions?.length || $unansweredRequiredInfoQuestions?.length)` should suffice